diff --git a/node_modules/expo-glass-effect/ios/GlassEffectModule.swift b/node_modules/expo-glass-effect/ios/GlassEffectModule.swift
index a82814a..aa6f2d2 100644
--- a/node_modules/expo-glass-effect/ios/GlassEffectModule.swift
+++ b/node_modules/expo-glass-effect/ios/GlassEffectModule.swift
@@ -6,18 +6,21 @@ public final class GlassEffectModule: Module {
   public func definition() -> ModuleDefinition {
     Name("ExpoGlassEffect")
 
-    Constant("isLiquidGlassAvailable") {
-      #if compiler(>=6.2)  // Xcode 26
-      if #available(iOS 26.0, tvOS 26.0, macOS 26.0, *) {  // iOS 26
-        if let infoPlist = Bundle.main.infoDictionary,
-          let requiresCompatibility = infoPlist["UIDesignRequiresCompatibility"] as? Bool {
-          // TODO(@uabx): Add a check for maximum SDK version when apple disables this flag
-          return !requiresCompatibility  // If the app requires compatibility then it will not use liquid glass
+    Constants {
+      let isAvailable: Bool = {
+        #if compiler(>=6.2)  // Xcode 26
+        if #available(iOS 26.0, tvOS 26.0, macOS 26.0, *) {  // iOS 26
+          if let infoPlist = Bundle.main.infoDictionary,
+            let requiresCompatibility = infoPlist["UIDesignRequiresCompatibility"] as? Bool {
+            // TODO(@uabx): Add a check for maximum SDK version when apple disables this flag
+            return !requiresCompatibility  // If the app requires compatibility then it will not use liquid glass
+          }
+          return true
         }
-        return true
-      }
-      #endif
-      return false
+        #endif
+        return false
+      }()
+      return ["isLiquidGlassAvailable": isAvailable]
     }
 
     View(GlassView.self) {
